tests/Tests.AzureAppConfiguration/KeyVaultReferenceTests.cs

```diff
@@ -284,7 +284,7 @@ public void ServerRequestIsMadeWhenDefaultCredentialIsSet()
             mockClient.Setup(c => c.GetConfigurationSettingsAsync(It.IsAny<SettingSelector>(), It.IsAny<CancellationToken>()))
                 .Returns(new MockAsyncPageable(new List<ConfigurationSetting> { _kv }));
 
-            KeyVaultReferenceException ex = Assert.Throws<KeyVaultReferenceException>(() =>
+            Assert.Throws<AuthenticationFailedException>(() =>
```

Comment:
>AuthenticationFailedException [](start = 26, length = 29)

This looks unrelated to the PR, but I'm assuming this test randomly started failing. The only way I can think of that happening was that it was missed in a previous PR, or we depend on a "latest-patch-version" of key vault client which now throws a different PR. 

Shouldn't we catch this new `AuthenticationFailedException` and convert it to `KeyVaultReferenceException` as we were doing before?

Reply:
I don't think its due to those reasons. I believe we had managed identity working for that specific key-vault in the past, so it would succeed with authentication but fail to find the secret and throw `KeyVaultReferenceException`. Now the managed identity endpoint doesn't seem to work, not sure why, so the test fails with `Azure.Identity.AuthenticationFailedException`.

Unfortunately, we can't handle this exception in the configuration provider since this belongs to the `Azure.Identity` package and we don't want a dependency on that package.

Reply:
As discussed offline, I have excluded this change from current PR and will send a PR for it separately along with other related changes.